mlir/include/mlir/Dialect/LLVMIR/LLVMIntrinsicOps.td

```diff
@@ -712,6 +712,25 @@ def LLVM_masked_compressstore
   let arguments = (ins LLVM_AnyVector, LLVM_AnyPointer, LLVM_VectorOf<I1>);
 }
 
+/// Create a call to trap intrinsic
+def LLVM_Trap : LLVM_ZeroResultIntrOp<"trap">;
+
+/// Create a call to debugtrap intrinsic
+def LLVM_DebugTrap : LLVM_ZeroResultIntrOp<"debugtrap">;
+
+/// Create a call to ubsantrap intrinsic
+def LLVM_UBSanTrap : LLVM_ZeroResultIntrOp<"ubsantrap"> {
+  let arguments = (ins I8Attr:$failureKind);
+  string llvmBuilder = [{
+    builder.CreateIntrinsic(
+      llvm::Intrinsic::ubsantrap, {}, {builder.getInt8($failureKind)});
+  }];
+  string mlirBuilder = [{
+    $_op =
+      $_builder.create<LLVM::UBSanTrap>($_location, $_int_attr($failureKind));
+  }];
```

Comment:
As the argument passed to this intrinsic is `immarg` (represented as attributes in MLIR), we need to define functions for `MLIR <-> LLVM` translation.